actionpack/lib/action_dispatch/journey/gtg/simulator.rb

```diff
@@ -22,18 +24,24 @@ def initialize(transition_table)
 
         def memos(string)
           input = StringScanner.new(string)
-          state = [0]
-          while sym = input.scan(%r([/.?]|[^/.?]+))
+          state = INITIAL_STATE
+
+          while sym = input.scan(MEMO_PATTERN)
             state = tt.move(state, sym)
           end
 
-          acceptance_states = state.find_all { |s|
-            tt.accepting? s
-          }
+          acceptance_states = state.each_with_object([]) do |s, memos|
+            next unless tt.accepting? s
+            memo = tt.memo(s)
 
-          return yield if acceptance_states.empty?
+            if memo.is_a?(Array)
+              memos.concat(memo)
+            elsif memo.present?
+              memos << memo
```

Comment:
When will pass the `elsif` branch?

Reply:
It was indeed unnecessary. The transition table memos default to an empty array. I changed the conditional to just verify if the memo is empty before concatenating.